misc/ls-extensions/modules/project-design-service/src/main/java/io/ballerina/projectdesign/generators/entity/EntityModelGenerator.java

```diff
@@ -70,42 +75,79 @@ public EntityModelGenerator(SemanticModel semanticModel, ComponentModel.PackageI
                 if (typeDefinitionSymbol.typeDescriptor() instanceof RecordTypeSymbol) {
                     String entityName = getEntityName(packageId, typeDefinitionSymbol.moduleQualifiedName());
-                    List<Attribute> attributeList = new ArrayList<>();
-                    List<String> inclusionList = new ArrayList<>();
                     RecordTypeSymbol recordTypeSymbol = (RecordTypeSymbol) typeDefinitionSymbol.typeDescriptor();
-                    Map<String, RecordFieldSymbol> recordFieldSymbolMap =
-                            getOriginalFieldMap(recordTypeSymbol, inclusionList, entityName);
-                    for (Map.Entry<String, RecordFieldSymbol> fieldEntry : recordFieldSymbolMap.entrySet()) {
-
-                        RecordFieldSymbol fieldEntryValue = fieldEntry.getValue();
-                        // when is the field name is optional? Tested with record inclusion and res types
-                        String fieldName = fieldEntryValue.getName().get(); // need to handle
-                        String fieldType = fieldEntryValue.typeDescriptor().signature();
-                        boolean optional = fieldEntryValue.isOptional();
-                        String defaultValue = ""; //need to address
-                        boolean nillable = isNillable(fieldEntryValue.typeDescriptor());
-                        List<Association> associations =
-                                getAssociations(fieldEntryValue.typeDescriptor(), entityName, optional, nillable);
-                        Attribute attribute =
-                                new Attribute(fieldName, fieldType, optional, nillable, defaultValue, associations,
-                                        getElementLocation(fieldEntryValue));
-                        attributeList.add(attribute);
-                    }
-
-                    Entity entity = new Entity(attributeList, inclusionList, getElementLocation(typeDefinitionSymbol));
-                    entities.put(entityName, entity);
+                    this.types.put(entityName, getType(recordTypeSymbol, entityName,
+                            getElementLocation(typeDefinitionSymbol), false));
                 }
             }
         }
-        return entities;
+        return types;
+    }
+
+    private Entity getType(RecordTypeSymbol recordTypeSymbol, String entityName, ElementLocation elementLocation,
+                           boolean isAnonymous) {
+        List<Attribute> attributeList = new ArrayList<>();
+        List<String> inclusionList = new ArrayList<>();
+        Map<String, RecordFieldSymbol> recordFieldSymbolMap =
+                getOriginalFieldMap(recordTypeSymbol, inclusionList, entityName);
+        for (Map.Entry<String, RecordFieldSymbol> fieldEntry : recordFieldSymbolMap.entrySet()) {
+            attributeList.add(getAttribute(fieldEntry.getValue(), entityName));
+        }
```

Comment:
Cant we iterate over the `recordFieldSymbolMap` values (instead of the entry set), since don't have a requirement to use the key?

Reply:
Yes. Updated